iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergMergeRecordReader.java

```diff
@@ -52,7 +55,14 @@ public void initialize(InputSplit split, TaskAttemptContext newContext) {
   }
 
   private CloseableIterator<T> nextTask() {
-    return openGeneric(mergeSplit.getContentFile(), getTable().schema()).iterator();
+    CloseableIterator<T> closeableIterator = openGeneric(mergeSplit.getContentFile(), getTable().schema()).iterator();
+    if (mergeSplit.getContentFile() instanceof DeleteFile) {
+      return new IcebergAcidUtil.MergeVirtualColumnAwareIterator<>(closeableIterator,
+              IcebergAcidUtil.createSerdeSchemaForDelete(getTable().schema().columns()), getConf(),
```

Comment:
nit, maybe extract serde into local var for readability

Reply:
Done.